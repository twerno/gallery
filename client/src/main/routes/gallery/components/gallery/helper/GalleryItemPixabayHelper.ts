import { IResponsivePictureProps } from "../GalleryResponsivePicture";
import { IPixabayImage } from "@shared/";

export default {
    convertToResponsivePictureProps
};

function convertToResponsivePictureProps(image: IPixabayImage): IResponsivePictureProps {
    const img_small = image.previewURL;
    const img_640 = image.webformatURL;
    const img_180 = img_640.replace(/_\d{3}\./, '_180\.');

    const alt = image?.tags?.split(',')
        .map<string>(s => s.trim())
        .reduce((prev, tag) => `${prev} ${tag}`, '')
        .trim();

    return {
        alt,
        defaultImgSrc: img_small,
        responsiveSrcList: [
            { src: img_180, media: '(min-width: 600px)' },
        ]
    };
}
