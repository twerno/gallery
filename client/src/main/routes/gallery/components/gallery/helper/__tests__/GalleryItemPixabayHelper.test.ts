import { IPixabayImage } from '@shared/';
import '@testing-library/jest-dom/extend-expect';
import GalleryItemPixabayHelper from '../GalleryItemPixabayHelper';

describe('GalleryItemPixabayHelper', () => {

    test('convertToResponsivePictureProps', async () => {
        const source: IPixabayImage = {
            pageURL: 'http://example.com.not/page',
            largeImageURL: 'http://example.com.not/large.jpg',
            previewURL: 'http://example.com.not/preview.jpg',
            id: 'id',
            previewHeight: '200',
            previewWidth: '200',
            tags: 'tag1, tag2',
            webformatURL: 'http://example.com.not/webformatURL.jpg',
        };

        const responsivePicture = GalleryItemPixabayHelper.convertToResponsivePictureProps(source);
        expect(responsivePicture).toMatchSnapshot();
    });
});

