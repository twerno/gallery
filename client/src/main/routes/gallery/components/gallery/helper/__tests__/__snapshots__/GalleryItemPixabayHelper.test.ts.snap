// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GalleryItemPixabayHelper convertToResponsivePictureProps 1`] = `
Object {
  "alt": "tag1 tag2",
  "defaultImgSrc": "http://example.com.not/preview.jpg",
  "responsiveSrcList": Array [
    Object {
      "media": "(min-width: 600px)",
      "src": "http://example.com.not/webformatURL.jpg",
    },
  ],
}
`;
